src/plugin-interfaces.ts

```diff
@@ -0,0 +1,163 @@
+import { Constants } from './constants';
+import { TraceLabels } from './trace-labels';
+
+export type Func<T> = (...args: any[]) => T;
```

Comment:
nit: use a more descriptive name perhaps.

Reply:
Not sure if there is anything more specific than `Func` (or `Function`) for something this broad.